phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -297,24 +297,24 @@ void closeConnection() throws SQLException {
         public Job getJob() throws Exception {
             if (isPartialBuild) {
                 return configureJobForPartialBuild();
-            } else {
+            } else if (useSnapshot || !useDirectApi || (!isLocalIndexBuild && pDataTable.isTransactional())) {
                 long maxTimeRange = pIndexTable.getTimeStamp() + 1;
                 // this is set to ensure index tables remains consistent post population.
-
                 if (pDataTable.isTransactional()) {
                     configuration.set(PhoenixConfigurationUtil.TX_SCN_VALUE,
                             Long.toString(TransactionUtil.convertToNanoseconds(maxTimeRange)));
                     configuration.set(PhoenixConfigurationUtil.TX_PROVIDER, pDataTable.getTransactionProvider().name());
                 }
                 configuration.set(PhoenixConfigurationUtil.CURRENT_SCN_VALUE,
                         Long.toString(maxTimeRange));
-                if (useSnapshot || !useDirectApi || (!isLocalIndexBuild && pDataTable.isTransactional())) {
-                    return configureJobForAysncIndex();
-                }
-                else {
-                    //Local and non-transactional global indexes to be built on the server side
-                    return configureJobForServerBuildIndex();
-                }
+                return configureJobForAysncIndex();
+            }
+            else {
+                // Local and non-transactional global indexes to be built on the server side
```

Comment:
If this is a direct build, then SCN is not set?

Reply:
Yes, if this is a direct build on non-transactional tables then SCN is not set